docusaurus/docs/create-a-plugin/extend-a-plugin/instrument-a-backend-plugin.md

```diff
@@ -0,0 +1,421 @@
+    ctxLogger.Info("Info msg", "queryType", "default")
+    ctxLogger.Warning("Warning msg", "key", "value")
+    ctxLogger.Error("Error msg", "error", errors.New("An error occurred"))
+}
+```
+
+#### Include additional contextual information in logs
+
+If you want to propagate additional contextual key/value pairs to subsequent code/logic you can use the [log.WithContextualAttributes](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/log#WithContextualAttributes) function.
+
+**Example:**
+
+```go
+func (ds *Datasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+    ctxLogger := ds.logger.FromContext(ctx)
+    ctxLogger.Debug("Debug msg", "someID", 1)
+    ctxLogger.Info("Info msg", "queryType", "default")
+    ctxLogger.Warning("Warning msg", "key", "value")
+    ctxLogger.Error("Error msg", "error", errors.New("An error occurred"))
+
+    for _, q := range req.Queries {
+        childCtx = log.WithContextualAttributes(ctx, []any{"refID", q.RefID, "queryType", q.QueryType})
+        ds.handleQuery(childCtx, q)
+    }
+}
+```
+
+### Best practices/recommendations
+
+- Start the log message with a capital letter, e.g. `logger.Info("Hello world")` instead of `logger.Info("hello world")`.
+- The log message should be an identifier for the log entry, try to avoid parameterization, e.g. `logger.Debug(fmt.Sprintf(“Something happened, got argument %d”, “arg”)), in favor of key-value pairs for additional data, e.g. `logger.Info(“Something happened”, “argument”, “arg”).
+- Prefer using camelCase style when naming log keys, e.g. remoteAddr or userID, to be consistent with Go identifiers.
+- Use the key `error` when logging Go errors, e.g. `logger.Error("Something failed", "error", errors.New("An error occurred")`.
+- Use a contextual logger whenever you have access to a `context.Context`.
+- Do not log sensitive information, such as data source credentials or IP addresses, or other personally identifiable information.
+
+#### Validate and sanitize input coming from user input
+
+If log messages or key/value pairs originate from user input they should be validated and sanitized. Be careful to not expose any sensitive information in log messages e.g. secrets, credentials etc. It's especially easy to do by mistake when including a Go struct as value.
+
+#### When to use which log level?
+
+- **Debug:** Informational messages of high frequency and/or less-important messages during normal operations.
+- **Info:** Informational messages of low frequency and/or important messages.
+- **Warning:** Should in normal cases not be used/needed. If used should be actionable.
+- **Error:** Error messages indicating some operation failed (with an error) and the program didn't have a way to handle the error.
+
+:::note
+
+Incoming requests of high frequency are normally more common for the QueryData endpoint since for example the nature of a dashboard generates a request per panel and/or query.
```

Comment:
```suggestion
Incoming requests of high frequency are normally more common for the `QueryData` endpoint, since--for example--the nature of a dashboard generates a request per panel and/or query.
```